app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -4,8 +4,7 @@ module MfaSetupConcern
   def confirm_user_authenticated_for_2fa_setup
     authenticate_user!(force: true)
     return if user_fully_authenticated?
-    return unless MfaPolicy.new(current_user).two_factor_enabled? &&
-                  !user_session[:signing_up]
```

Comment:
This `:signing_up` flag was in the session to allow the backup code only flow. I pulled it out here and a bunch of other places we no longer needed.